### DISCUSSION TITLE
Wasseruhr Werte Frage

### DISCUSSION THREAD

**Discussion by AdmiralRaccoon** [jomjol/AI-on-the-edge-device#1461] [General]
Moin, ich habe derzeit folgenden Wert auf der Uhr.
Für Ai On The Edge ist es ein Fehler, weil der Wert der 1. Nachkommastelle von 9 auf 0 gerutscht ist. (Was er ja auch richtig gelesen hat)
Kann man in dem Fall überhaupt  was machen?

![image](https://user-images.githubusercontent.com/41727850/205254894-f067bcb1-1116-4f36-be3b-94d3946d12ef.png)

**Comment by Triffid-ger** [comment]
Hi,
doch, das sollte sich lösen lassen, wenn du unter "Settings/Configuration" oben den Haken bei "Expert Mode" setzt und dann weiter unten PreValueUse auf "true" und CheckDigitIncreaseConsistency auf "true" setzt.
Manchmal hat er beim Zehnerübergang trotzdem noch Probleme (zumindest in meinem Setup mit etwas veralteten tflite-Dateien) aber ansonsten sollte er den Wert nun korrekt ermitteln.

**Reply by AdmiralRaccoon** [reply]
Danke für den Tipp, habe ich jetzt mal so wie unten eingestellt, aber hat leider das gleiche Resultat.
Vielleicht fehlt da noch ein Liter bevor es wieder geht. :-D

![image](https://user-images.githubusercontent.com/41727850/205288247-389e7b81-928b-4feb-bde6-131df4c0b291.png)
![image](https://user-images.githubusercontent.com/41727850/205288325-6b4eb1ed-082f-4b71-beea-291b48eb3c73.png)

**Comment by Triffid-ger** [comment]
Hmmm. Ok. Ich vermute, du verwendest eine aktuelle tflite-Datei für den digitalen Bereich? Also sowas wie "dig-cont_xxx-s3.tflite"?
Das ist zwar die aktuell empfohlene Vorgehensweise, weil auch durchrollende Ziffern erkannt werden, aber bei mir funktioniert das auch nicht sonderlich gut.
Das Problem scheint hier zu sein, dass das System sieht, dass die 70 m³ noch nicht ganz angezeigt werden und deswegen wirft es den Wert 69 aus. Bei deiner Wasseruhr (wie auch bei meiner) stellen sich die Ziffern aber erst nach dem 0-Durchgang auf den korrekten Folgewert ein. Besser wäre es in diesem Fall, die Erkennung schlüge fehl, dann würde die Logik bei einem Raw-Wert von 000NN.0178 und einem Pre-Value von 00069.9321 erkennen, dass da der nächste volle m³-Wert erreicht wurde und richtig auf die 70.0178 kommen.

Ich will nun nicht dazu raten, Rückschritte zu machen und vielleicht kennt jemand mit mehr Ahnung als ich auch eine bessere Lösung, aber bei mir läuft alles mit einem geradezu beunruhigenden Grad an Perfektion, seit ich in der Config.ini beim Digitalmodell "dig-class100-0130_s2_q.tflite" eingestellt habe.
Eventuell würde ich vorher aber mal bis zum nächsten Übergang warten, ob es dann korrekt läuft. Also öfter mal ein Vollbad nehmen. Kann sowieso nicht schaden, bei dem Wetter.
